core/src/main/scala/filodb.core/binaryrecord2/RecordSchema.scala

```diff
@@ -462,6 +462,30 @@ object RecordSchema {
   final def schemaID(bytes: Array[Byte]): Int =
     if (bytes.size >= 6) schemaID(bytes, UnsafeUtils.arayOffset) else -1
 
+  /**
+    * mutate the dataschema of a partitionKey.
+    * Used during downsampling to mutate the resulting partkey's dataschema esp. for Guages.
+    *
+    * @throws java.util.NoSuchElementException if there is not downsample schema
+    */
+  final def updateDSSchema(addr: BinaryRegion.NativePointer, schemas: Schemas): Unit = {
+    val dsSchemaId = schemas(schemaID(addr)).downsample.get
+    UnsafeUtils.setShort(UnsafeUtils.ZeroPointer, addr + 4, dsSchemaId.schemaHash.toShort)
+  }
+
+  /**
+    * Build a partkey from the source partkey and change the downsample schema.
+    * Useful during downsampling as dataschema may differ.
+    *
+    * @throws java.util.NoSuchElementException if there is not downsample schema
+    */
+  final def buildDSPartKey(pkByte: Array[Byte], schemas: Schemas): Array[Byte] = {
```

Comment:
Can we expand `DS` to `Downsample` in both method names?

Reply:
fixed